dfanalyzer/graph_visualization/cytoscape.py

```diff
@@ -0,0 +1,352 @@
+import json
+import networkx as nx
+import ipycytoscape
+import ipywidgets as widgets
+from IPython.display import display
+# import ipycytoscape
+# import networkx as nx
+
+
+class CytoGraph:
+    
+    def get_json(self, graph, degree_dict = {}):
+        '''
+        Input: networkx graph,  dict (key = node_id and value = degree)
+        Returns: json_data (format needed for ipycytoscape visualization)
+        '''
+        # for pid_tid -> file bipartite graphs
+        json_data = {
+                    'nodes': [{'data': {'id': str(node[0]), 'degree': int(degree_dict[node[0]])}} for node in graph.nodes(data=True)],
+                    'edges': [{'data': {'source': str(edge[0]), 'target': str(edge[1]), 'directed': True}} for edge in graph.edges(data=True)]}
+
+        return json_data
+    
+    
+    def get_rich_json(self, graph, degree_dict = {}):
+        '''
+        Get json with Nodes and Edges attributes
+        Input: networkx graph,  dict (key = node_id and value = degree)
+        Returns: json_data (format needed for ipycytoscape visualization)
+        '''
+        # for pid_tid -> file bipartite graphs
+        # json_data = {
+        #     'nodes': [{'data': {'id': str(node[0]),'degree': int(degree_dict[str(node[0])]), 'bipartite': int(node[1]['bipartite'])}} for node in graph.nodes(data=True)],
+        #     'edges': [{'data': {'source': str(edge[0]), 'target': str(edge[1]), 'weight': int(edge[2]['weight'])}} for edge in graph.edges(data=True)]}
+
+        # # for unweighted unipartite graphs
+        # json_data = {
+        #     'nodes': [{'data': {'id': str(node[0]), 'degree': int(degree_dict[node[0]])}} for node in graph.nodes(data=True)],
+        #     'edges': [{'data': {'source': str(edge[0]), 'target': str(edge[1]), 'directed': True}} for edge in graph.edges(data=True)]}
+
+        # for directed
+        # Prepare nodes and edges data
+        nodes = [{'data': {'id': str(node), 'label': str(node), 'outdegree': int(graph.out_degree(node))}} for node in graph.nodes()]
+        edges = []
+        for edge in graph.edges(data=True):
+            u , v, t = edge[0], edge[1], edge[2]['wt']
+            edge_data = {'data': {'source': str(u), 'target': str(v), 'wt': int(t)}}
+            if graph.has_edge(v, u):
+                edge_data['classes'] = 'bidirectional'
+            edges.append(edge_data)
+        
+        json_data = {'nodes': nodes, 'edges': edges}
+
+        return json_data
+
+    def get_style(self, style_name):
+        '''
+        Input: style_name (string as filter identifier)
+        Output: style  for cytoscape visualization
+        '''
+        if style_name == 'default':  # this will check if the filters are working properly
+            style = [
```

Comment:
Could this style be moved to a json file in `dfanalyzer/graph/style/<style_name>.json`